src/cunumeric/index/advanced_indexing.cc

```diff
@@ -0,0 +1,109 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "cunumeric/index/advanced_indexing.h"
+#include "cunumeric/index/advanced_indexing_template.inl"
+
+namespace cunumeric {
+
+using namespace Legion;
+using namespace legate;
+
+template <LegateTypeCode CODE, int DIM1, int DIM2, bool IS_SET>
+struct AdvancedIndexingImplBody<VariantKind::CPU, CODE, DIM1, DIM2, IS_SET> {
+  using VAL = legate_type_of<CODE>;
+
+  void compute_output(Buffer<VAL>& out,
+                      const AccessorRO<VAL, DIM1>& input,
+                      const AccessorRO<bool, DIM2>& index,
+                      const Pitches<DIM1 - 1>& pitches_input,
+                      const Rect<DIM1>& rect_input,
+                      const Pitches<DIM2 - 1>& pitches_index,
+                      const Rect<DIM2>& rect_index,
+                      int volume) const
```

Comment:
I think `volume` should be `size_t` everywhere. Currently in some places it's `size_t` and in others it's `int`.